src/javascript/app/pages/user/new_account/welcome_page.js

```diff
@@ -1,45 +1,59 @@
+            init();
             if (Client.hasAccountType('real')) {
                 window.location.href = Client.defaultRedirectUrl();
                 showLoadingImage(el_welcome_container, 'dark');
             }
 
-            const upgrade_info = Client.getUpgradeInfo();
-            const el_upgrade_title = getElementById('upgrade_title');
-            let upgrade_btn_txt = '';
+            getElementById('default').addEventListener('click', () => {
+                /* eslint no-unused-expressions: ["error", { "allowTernary": true }] */
+                (getCanUpgrade('iom') || (is_uk && is_unwelcome_uk))
+                    ? window.location.href = Url.urlFor('/new_account/realws')
+                    : window.location.href = Client.defaultRedirectUrl();
+            });
 
-            if (upgrade_info.can_upgrade_to.length > 1) {
-                upgrade_btn_txt = localize('Real Account');
-            } else if (upgrade_info.can_upgrade_to.length === 1) {
-                upgrade_btn_txt = upgrade_info.type[0] === 'financial'
-                    ? localize('Financial Account')
-                    : localize('Real Account');
-            }
-            el_upgrade_title.html(upgrade_btn_txt);
-            el_welcome_container.setVisibility(1);
-
-            const upgrade_url = upgrade_info.can_upgrade_to.length > 1
-                ? 'user/accounts'
-                : Object.values(upgrade_info.upgrade_links)[0];
-
-            if (upgrade_info.can_upgrade) {
-                const upgrade_btn = getElementById('upgrade_btn');
-                if (upgrade_btn) {
-                    upgrade_btn.html(createElement('span', { text: localize('Upgrade now') })).setAttribute('href', Url.urlFor(upgrade_url));
-                    upgrade_btn.classList.remove('button-disabled');
+            getElementById('cfd').addEventListener('click', () => {
+                if (is_virtual && upgrade_info.can_upgrade_to.length) {
+                    if (getCanUpgrade('svg')){window.location.href = Url.urlFor('/user/metatrader'); return;}
+                    if (getCanUpgrade('maltainvest')){window.location.href = Client.defaultRedirectUrl(); return;}
+                    if (getCanUpgrade('iom') && (is_uk && is_unwelcome_uk))window.location.href = Url.urlFor('/user/metatrader');
                 }
-            }
+            });
+
+            getElementById('d_ptions').addEventListener('click', () => {
+                if (is_virtual && upgrade_info.can_upgrade_to.length){
+                    if (getCanUpgrade('svg'))window.location.href = Client.defaultRedirectUrl();
+                    if (getCanUpgrade('maltainvest'))window.location.href = Url.urlFor('new_account/digital_options');
+                    if (getCanUpgrade('iom') && (is_uk && is_unwelcome_uk))window.location.href = Url.urlFor('/new_account/realws');
+                }
+            });
```

Comment:
where are we removing all the event listeners? they should be removed on onunload function or before the add event

Reply:
done also